jit: Handle EBB information correctly

qwe661234 opened the pull request:
1. When BB is replaced by cache, the EBB information recorded in other BB is not updated. This could result in outdated EBB information, and the replaced BB might have been created in a different memory location. To address this issue, we should verify whether the EBB information is outdated or not and update it accordingly each time we use EBB information.
2. Some non-brnach instruction can be the end of a BB when a BB is fill, so we modify insn_is_unconditional_branch into insn_is_conditional_branch that we can only extend BB when its end is conditional branch instruction.

Solve: #173

jserv commented on src/emulate.c:
```diff
@@ -195,20 +195,18 @@ static inline bool insn_is_branch(uint8_t opcode)
 }
 
-static inline bool insn_is_unconditional_branch(uint8_t opcode)
+static inline bool insn_is_conditional_branch(uint8_t opcode)
```
You shall explain the motivation and considerations for such modifications.

qwe661234 committed:
jit: Handle EBB information correctly

1. When BB is replaced by cache, the EBB information recorded in other BB
is not updated. This could result in outdated EBB information, and the replaced
BB might have been created in a different memory location. To address this
issue, we should verify whether the EBB information is outdated or not and
update it accordingly each time we use EBB information.
2. Some non-brnach instruction can be the end of a BB when a BB is fill,
so we modify insn_is_unconditional_branch into insn_is_conditional_branch
that we can only extend BB when its end is conditional branch instruction.

Solve: #173
